src/include/test/unit_api_flat_l2_index.cc

```diff
@@ -43,7 +43,7 @@ TEST_CASE("api_flat_l2_index: test test", "[api_flat_l2_index]") {
 
 TEST_CASE("api: flat_l2_index", "[api][flat_l2_index]") {
   tiledb::Context ctx;
-  auto a = IndexFlatL2(ctx, fmnist_inputs_uri);
+  auto a = IndexFlatL2(ctx, fmnist_inputs_uri.string());
```

Comment:
Again, we should have an overload, rather than requiring explicit conversion to string.